### PR TITLE
Fix API review warnings in for autoconfigure.keyvault and autoconfigure.messaging

### PR DESCRIPTION
**Pull request by chenrujun**
As title.

Fix warnings in api-review page: https://apiview.dev/Assemblies/Review/783a66f61ccd4452a67fbebfafb6fcd3#com.azure.spring.cloud.autoconfigure.keyvault.certificates

This PR is only for  these packages: 
1. `com.azure.spring.cloud.autoconfigure.keyvault`.
2. `com.azure.spring.cloud.autoconfigure.messaging`

### CODE REVIEW

**Comment by chenrujun** [comment]
Hi, @saragluna , please help to review this PR when you have time.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/AzureKeyVaultCertificateAutoConfiguration.java [hunk 1]
```diff
@@ -21,13 +22,14 @@
...
+ * {@link EnableAutoConfiguration Auto-configuration} for Azure Key Vault Certificate support.
+ *
+ * @since 4.0.0
  */
 @ConditionalOnClass(CertificateClientBuilder.class)
 @ConditionalOnExpression("${spring.cloud.azure.keyvault.certificate.enabled:true}")
 @ConditionalOnProperty("spring.cloud.azure.keyvault.certificate.endpoint")
-public class AzureKeyVaultCertificateAutoConfiguration extends AzureServiceConfigurationBase {
-
+public final class AzureKeyVaultCertificateAutoConfiguration extends AzureServiceConfigurationBase {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/AzureKeyVaultCertificateAutoConfiguration.java:32]: There's no need to mark the class final here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java [hunk 2]
```diff
@@ -8,18 +8,28 @@
 import com.azure.spring.service.keyvault.certificates.KeyVaultCertificateProperties;
 
 /**
- * Properties for Azure Key Vault Certificate.
+ * Azure Key Vault Certificate properties.
+ *
+ * @since 4.0.0
  */
-public class AzureKeyVaultCertificateProperties extends AzureKeyVaultProperties implements KeyVaultCertificateProperties {
+public final class AzureKeyVaultCertificateProperties extends AzureKeyVaultProperties implements KeyVaultCertificateProperties {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java:15]: same here

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java [hunk 3]
```diff
@@ -8,18 +8,28 @@
...
 
     public static final String PREFIX = "spring.cloud.azure.keyvault.certificate";
 
     private CertificateServiceVersion serviceVersion;
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java:19]: also add java doc here

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/env/KeyVaultEnvironmentPostProcessor.java [hunk 4]
```diff
@@ -29,13 +29,19 @@
 
 /**
  * Leverage {@link EnvironmentPostProcessor} to add Key Vault secrets as a property source.
+ *
+ * @since 4.0.0
  */
-public class KeyVaultEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered {
+public final class KeyVaultEnvironmentPostProcessor implements EnvironmentPostProcessor, Ordered {
```
Comment (saragluna) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/env/KeyVaultEnvironmentPostProcessor.java:35]: no need here

**Review by saragluna — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/AzureKeyVaultCertificateAutoConfiguration.java (hunk 1 above)
Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/AzureKeyVaultCertificateAutoConfiguration.java:32]: It won't cause any problem if we add `final`, how about keep it?

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java (hunk 3 above)
Comment (chenrujun) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java:19]: This is a private field, no need to add java doc here. And here is no related requirement in api-review page.

**Comment by chenrujun** [comment]
@saragluna ping.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/AzureKeyVaultCertificateAutoConfiguration.java (hunk 1 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/AzureKeyVaultCertificateAutoConfiguration.java:32]: I think we've chosen to follow Spring Boot's style, and to be consistent for all our classes in this autoconfigure module, I prefer not adding the final here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java (hunk 3 above)
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/java/com/azure/spring/cloud/autoconfigure/keyvault/certificates/properties/AzureKeyVaultCertificateProperties.java:19]: This is for configuration properties metada.

**Comment by chenrujun** [comment]
@saragluna , updated.

**Review by saragluna — APPROVED** [review]
LGTM.

**Pull request merged**